cmd/start.go

```diff
@@ -71,15 +73,43 @@ var startCmd = &cobra.Command{
 	Long:  ``,
 	Run: func(cmd *cobra.Command, args []string) {
 		// Configure loggers -------------------------------------------------------
-		log.SetFormatter(&log.JSONFormatter{})
-		log.SetOutput(os.Stdout)
+		var level zapcore.Level
+		var err error
 		if Debug {
-			log.SetLevel(log.DebugLevel)
+			level = zapcore.DebugLevel
 		} else {
-			log.SetLevel(log.InfoLevel)
+			level = zapcore.InfoLevel
 		}
+		cfg := zap.Config{
```

Comment:
Minor thing - per my understanding, we mostly use default zap logger configurations, except for log level. Hence, what if `NewLogger(logger)` contract change to `NewLogger(zapcore.Level)` and internally perform the default loading ? We could of course expose `NewLogger(zap.Config)` if we really needs to :) IMO this makes `start.go` little cleaner.

Reply:
Great suggestion, i'll introduce this change

Reply:
Ive moved the generation of the logger into a separate function, this way we can still pass nil loggers into the NewLogger function, keeping both functions very simple